### ISSUE TITLE
Patch does not work for zoom sdk version 5.14.10.17290

### ISSUE THREAD

**Issue by syedmub** [aws-samples/aws-virtual-participant-framework-for-rtc#21]
### Expected behavior and actual behavior

[Patch](https://github.com/aws-samples/aws-virtual-participant-framework-for-rtc/blob/main/virtual-participant-orchestrator-for-zoom-meeting/src/zoom_sdk_demo_v2.patch) included in the repo works for zoom windows SDK version `zoom-sdk-windows-5.12.8.10282`

Need a patch that works for latest windows sdk version i.e `5.14.10.17290` to utilise latest features from the SDK like enabling auto recording request when joining a meeting. It's mentioned [here](https://developers.zoom.us/docs/meeting-sdk/windows/raw-data/#before-you-start) that we need sdk version 5.13+ to request recording permission from host

### Steps to reproduce the problem

Apply this [patch](https://github.com/aws-samples/aws-virtual-participant-framework-for-rtc/blob/main/virtual-participant-orchestrator-for-zoom-meeting/src/zoom_sdk_demo_v2.patch) on to latest sdk version and it doesn't work.

**Comment by sinasojoodi** [comment]
A new patch is in the _20230608-zoom-sdk-v5-14-update_ branch. waiting for Zoom team to fix an unwanted modal popup stopping VPF bots from joining meetings. This seems to be regression that is introduced after version 5.12

**Comment by venkat-jeyaraman** [comment]
Hi @sinasojoodi any update this patch fix for latest version of Zoom SDK?. Thanks.

**Comment by sinasojoodi** [comment]
Unfortunately the patch has not yet been submitted by the Zoom team. I don't believe there is a technical blocker, rather it's behind other priorities. The current SDK version meets the minimum required until Nov 2023. My hope is the patch can be submitted by end of Sept.

**Comment by venkat-jeyaraman** [comment]
Hi @sinasojoodi thanks for the response. So at the moment with your example of 5.12.8.10282 it works fine but just that host to explicitly give permission to vpf user to enable local recording. But rather want to move to latest to enable auto recording with join token.

**Comment by sinasojoodi** [comment]
That is correct. Upgrading to 5.14 is treated as a feature enhancement until Nov when it becomes mandatory.


On Aug 18, 2023, at 4:36 AM, Venkatraman Jeyaraman ***@***.***> wrote:

﻿

Hi @sinasojoodi<https://github.com/sinasojoodi> thanks for the response. So at the moment with your example of 5.12.8.10282 it works fine but just that host to explicitly give permission to vpf user to enable local recording. But rather want to move to latest to enable auto recording with join token.

**Comment by venkat-jeyaraman** [comment]
@sinasojoodi I am trying to run the windows app locally by following your instructions in src/README.md file. In that instructions are given for 'Choose **"Release"** and **"x64"** configuration and build solution.' But in zoom windows sdk app documentation 
'Solution Configuration Debug and Solution Platform x64 are not supported at the moment.'. X64 is not supported at all. Just want to know how you managed to run on X64 platform.
Link: https://developers.zoom.us/docs/meeting-sdk/windows/get-started/build-run-play/#play-with-your-app

**Comment by sinasojoodi** [comment]
Unfortunately I don not have  a good answer since it just works for me. I do see the documentation from Zoom and I knew about "debug" not working but "x64" not being supported is news for me. I have always run the project  targeting "release" and "x64" and it just works.

Are you running into an issue? If so please post the specific issue on Zoom Dev Forum [1]. That's where Zoom SDK experts and Dev evangelists monitor and can look into it.

[1] https://devforum.zoom.us/c/meeting-sdk/windows/9
On Aug 19, 2023, at 7:11 PM, Venkatraman Jeyaraman ***@***.***> wrote:

﻿

@sinasojoodi<https://github.com/sinasojoodi> I am trying to run the windows app locally by following your instructions in src/README.md file. In that instructions are given for 'Choose "Release" and "x64" configuration and build solution.' But in zoom windows sdk app documentation
'Solution Configuration Debug and Solution Platform x64 are not supported at the moment.'. X64 is not supported at all. Just want to know how you managed to run on X64 platform.
Link: https://developers.zoom.us/docs/meeting-sdk/windows/get-started/build-run-play/#play-with-your-app

**Comment by venkat-jeyaraman** [comment]
hi @sinasojoodi thanks for the response. Finally  i got it running but when i allow the bot the record files the code throws an exception 'Unhandled exception thrown: read access violation.
type_instance->g_class was 0xFFFFFFFFFFFFFFFF'  from goobject-2.0.0.dll in SetupGstPipeline function.
Have you seen anything like that?.

**Comment by sinasojoodi** [comment]
No I have not. Would you kindly restart your env and log it as an new issue on GitHub.

Also does this happen with the 5.12 version of the SDK currently on the `main` branch? It would require downloading a different version of the SDK and the patch provided on that branch.


On Aug 20, 2023, at 1:52 AM, Venkatraman Jeyaraman ***@***.***> wrote:

﻿

hi @sinasojoodi<https://github.com/sinasojoodi> thanks for the response. Finally i got it running but when i allow the bot the record files the code throws an exception 'Unhandled exception thrown: read access violation.
type_instance->g_class was 0xFFFFFFFFFFFFFFFF' from goobject-2.0.0.dll in SetupGstPipeline function.
Have you seen anything like that?.

**Comment by sinasojoodi** [comment]
Would you kindly open a new GitHub issue and lable as a "question" and/or "discussion"?  This way the team can track progress of 5.14 feature branch more effectively and other experts can address your question as a discussion topic or a new feature request.

Thank you


On Aug 22, 2023, at 10:32 AM, Venkatraman Jeyaraman ***@***.***> wrote:

﻿

hi @sinasojoodi<https://github.com/sinasojoodi> Thanks for the response. I will raise a new issue with that screenshot. Unfortunately am not able to record on my computer, but surprisingly inside the docker container in ECS it works. I also have a question my aim is to combine audio and video and create mp4 out of it. Can i write 2 gstreamer pipelines in same windows application. One to send audio to KVS and other to combine audio and video and write to s3. Any thoughts would be appreciated here.

**Comment by osamabinsaleem** [comment]
Any update on this issue? @sinasojoodi Do you have any idea when it might get approved? I'm using this code for development and just got an email from zoom that I need to update the sdk package.

**Comment by venkat-jeyaraman** [comment]
@sinasojoodi excellent. Waiting to try it out on latest SDK.

**Comment by osamabinsaleem** [comment]
@sinasojoodi Any timelines on this? We're eagerly waiting to try the latest SDK. Thanks

**Comment by sinasojoodi** [comment]
Code is tested and working on the `develop` branch.  I will be pushing it into main after some cleanup to READMEs, some NPM dependencies for CI, and seeing if there is a short path to upgrade to 5.16 version of the Zoom SDK.  That will hopefully avoid forcing everyone to upgarde to a new version again in Feb 2024.

**Comment by sinasojoodi** [comment]
@osamabinsaleem @venkat-jeyaraman while the join meeting issue seems to have been resolved by the latest contribution from Max, we have discovered a new incompatibility with the 5.14.10 version of the SDK.  While the v5.14 sample app works locally, the app does not call the onMixAudio callbacks when running in a container. I've reached out to the Zoom SDK engineering for help. I will keep you posted on this thread. 

If you are testing locally, the code in `develop` branch should work with v5.14 of the SDK.

**Comment by osamabinsaleem** [comment]
@sinasojoodi Thanks for the update. Looking forward to hear the next update from you.

**Comment by venkat-jeyaraman** [comment]
@sinasojoodi thanks for the update. Will wait for your merge.In the mean time i will test it on develop branch locally.

**Comment by osamabinsaleem** [comment]
@sinasojoodi any update on this?

**Comment by sinasojoodi** [comment]
We finally had a  breakthrough @osamabinsaleem. While 5.14.10.17920 continues to challenge us with end-to-end audio streaming function, we were able to get v5.13.10.13355 to work end to end with a new patch that I just tested end to end.  This version of the SDK won't go out of date until Feb 3rd, 2024 and the Zoom team is well aware of the issue.  We will have plenty of time to address likely will skip forward to 5.16.x for Feb 2024.

Tonight I'm going to update the development branch to make the project compatible with Windows SDK v5.13. Since I've already tested it end to end and it is a critical fix, I will break-glass and promote to main.  I'll update this issue once complete

**Comment by osamabinsaleem** [comment]
Thanks a lot @sinasojoodi

**Comment by venkat-jeyaraman** [comment]
Thanks @sinasojoodi
